src/hooks/useOrders.ts

```diff
@@ -27,19 +28,130 @@ function filterDeletedOrders(orders: AuctionElement[]): AuctionElement[] {
   )
 }
 
-export function useOrders(): AuctionElement[] {
+interface Result {
+  orders: AuctionElement[]
+  forceOrdersRefresh: () => void
+  isLoading: boolean
+}
+
+export function useOrders(): Result {
   const { userAddress, networkId, blockNumber } = useWalletConnection()
   const [orders, setOrders] = useSafeState<AuctionElement[]>([])
+  const [offset, setOffset] = useSafeState<number | undefined>(0)
+  const [isLoading, setIsLoading] = useSafeState<boolean>(false)
+
+  // Tracking last seen values to decide whether to update
+  const [lastBlockNumber, setLastBlockNumber] = useSafeState<number | undefined>(undefined)
+  const [lastNetworkId, setLastNetworkId] = useSafeState<number | undefined>(undefined)
+  const [lastUserAddress, setLastUserAddress] = useSafeState<string | undefined>(undefined)
+
+  const _fetchOrders = useCallback(
+    async (
+      userAddress: string | undefined,
+      networkId: number | undefined,
+      offset: number | undefined,
+      areThereOrders?: boolean,
+    ): Promise<void> => {
+      if (userAddress === undefined || networkId === undefined || offset === undefined) {
+        return
+      }
+
+      await exchangeApi.getOrdersPaginated({ userAddress, networkId, offset }).then(({ orders, nextIndex }) => {
+        if (orders.length) {
+          // Apply filters (remove deleted orders)
+          const filteredOrders = filterDeletedOrders(orders)
+
+          // Store new orders, if any
+          if (offset === 0) {
+            // fresh start/refresh: replace whatever is stored
+            setOrders(filteredOrders)
+          } else if (filteredOrders.length) {
+            // incremental update: append
+            setOrders(curr => curr.concat(filteredOrders))
+          }
+        } else if (offset === 0 && areThereOrders) {
```

Comment:
Sadly, no.
If I use `orders` from the outer scope, there are two possibilities:
1) The value of `orders` will always be the same -- if using no deps on useCallback -- making it useless
2) The callback will be updated on every change of `orders`, and since the function is recursive, that is not a good thing.

The first version was just the way you describe and took me awhile to understand why it was failing

Reply:
okok